### PR TITLE
Add documentation for a development workflow for Kedro Projects on Databricks

### LINKED ISSUE #2445: Implement and document a hybrid IDE and Databricks Workspace workflow for developing Kedro projects for Databricks
## Description
This issue is a child of https://github.com/kedro-org/kedro/issues/2185. Implement and document the hybrid workflow described in the ticket https://github.com/kedro-org/kedro/issues/2282.

### PR DESCRIPTION
**Pull request by jmholzer**
## Description
Resolves #2445

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/integrations/databricks_development_workflow.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
+# Databricks development workflow
+
+This guide demonstrates a development workflow for Kedro projects on Databricks using Databricks Repos. This hybrid workflow involves developing your Kedro project using a local Integrated Development Environment (IDE) and Databricks notebooks for testing purposes. This guide demonstrates this workflow using an example based on the Kedro PySpark Iris starter.
+This workflow has several advantages, enabling:
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:4]: Advantages over what?

File: docs/source/integrations/databricks_development_workflow.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+- Auto-completion and suggestions for code, improving your development speed and accuracy.
+- Linters like Pylint or Flake8 can be integrated to catch potential issues in your code.
+- Static type checkers like Mypy can check types in your code, helping to identify potential type-related issues early in the development process.
+
+To set up these features, look for instructions specific to your IDE (for instance, [VS Code](https://code.visualstudio.com/docs/python/linting)).
+
+## Prerequisites
+
+- New or existing [AWS account](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/) with administrative privileges.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:20]: Administrative privileges for what? It seems unlikely that our users (especially QB on client side) would have full admin rights for an AWS account. What's the specific rights they need to have to follow this flow?

File: docs/source/integrations/databricks_development_workflow.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+- Active [Databricks deployment](https://docs.databricks.com/getting-started/index.html) on AWS (Databricks Community Edition won't suffice as it doesn't allow you to provision personal tokens).
+- A [Databricks cluster](https://docs.databricks.com/clusters/configure.html) configured with a recent version (>= 11.3 is recommended) of the Databricks runtime.
+- [Conda installed](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) on your local machine.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:23]: Does it have to be conda or is any virtual environment okay? And is this essential for this workflow or more of a best practice in general?

File: docs/source/integrations/databricks_development_workflow.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+```bash
+...
+[08/09/22 11:23:30] INFO     Model has accuracy of 0.960 on test data.                                        nodes.py:74
+                    INFO     Saving data to 'metrics' (MetricsDataSet)...                             data_catalog.py:382
+                    INFO     Completed 3 out of 3 tasks                                           sequential_runner.py:85
+                    INFO     Pipeline execution completed successfully.                                      runner.py:89
+```
+
+### Make local changes in your local development environment
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:190]:
```suggestion
### Make changes in your local development environment
```

File: docs/source/integrations/databricks_development_workflow.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+```bash
+git add -u
+git commit -m "Modify 1-nearest neighbour classifier to 3-nearest-neighbour classifier"
+git push origin --set-upstream feat/modified-classifier
+```
+
+#### Pull the new branch and check it out on Databricks Repos
+
+Navigate to your Databricks Repo in the Databricks Workspace UI. To pull the changes from your remote repository, look for the `Pull` button in the upper-right corner of the Repo page. Click it to sync the latest changes from the remote repository into your Databricks Repo.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:272]: It would be useful to add a picture here of the pull button and it's position you mention.

File: docs/source/integrations/databricks_development_workflow.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+
+Next, check the available branches. Click on the "Branches" button in the upper-left corner of the Repo page. You should see the newly pulled branch `feat/modified-classifier` in the list. Check out the `feat/modified-classifier` branch by clicking on it in the drop-down list. Your active branch is now feat/modified-classifier.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:274]: Same here, some visuals would be helpful.

File: docs/source/integrations/databricks_development_workflow.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+```bash
+git push origin
+```
+
+To sync the main branch to your Databricks Repo, navigate to your Databricks Repo in the Databricks Workspace UI and use the `Pull` button as you did previously.
+
+```{note}
+In a project with collaborators, the best practice for merging your work with the main branch is to open a pull request (PR) using your Git provider's interface. PRs enable your collaborators to review your work before merging it.
+```
+
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:319]: I think it would be useful to mention what steps users would need to repeat when cluster/processes have been restarted. E.g. the data upload to DBFS is ephemeral right?

**Review by merelcht — COMMENTED** [review]
I've done an initial read and left some questions and comments. I will go through the guide again and try it out. 

Some thoughts I had while reading this:

- There's a lot of steps. Perhaps it would be good to add these in the introduction so anyone reading this know what's going to be demonstrated.
- There's a bit of a mix of Databricks specific recommendation and just general development, e.g. why use an IDE and creating a conda environment. Personally, I would try and keep the recommendations for a good workflow to specific Databricks things and not go into how to properly work with git etc.

File: docs/source/integrations/databricks_development_workflow.md (hunk 2 above)
Comment (merelcht) [reply docs/source/integrations/databricks_development_workflow.md:20]: From the rest of the tutorial it's not clear to me why you need admin privileges? There's no mention of specific configurations or permissions.

File: docs/source/integrations/databricks_development_workflow.md [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+This command will create a new Kedro project using the PySpark Iris starter template. Initialize Git for the Kedro project by running the following commands:
+
+```bash
+cd iris-databricks
+git init
+```
+
+### Add your project to Version Control
+
+Create a new private Git repository with your Git provider. Then, generate an access token to connect your Kedro project with Databricks. Look up the documentation of your Git provider to learn how to generate an access token if you are unclear, as this is different for each provider.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:70]: I didn't initially realise these were separate steps. Perhaps put them on new lines to make that clear. 
I'm also not quite sure where to add my access token after I've created it? -> You've described this later on so maybe just remove this sentence here.

File: docs/source/integrations/databricks_development_workflow.md [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+This command will install the dependencies listed in your requirements.txt file to the Databricks cluster attached to your notebook.
+
+#### Upload project data to DBFS
+
+In this demonstration project, you will upload the project data to DBFS as the PySpark Iris starter comes with a catalog set up to access data stored on DBFS.
+
+There are several ways to upload data to DBFS. In this guide, it is recommended to use [Databricks CLI](https://docs.databricks.com/dev-tools/cli/dbfs-cli.html) because of the convenience it offers. Use the following Databricks CLI command to upload your data:
+
+```bash
+databricks fs cp --recursive data/ dbfs:/root/projects/iris-databricks/data
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:132]: Where should you run this command: in the notebook or locally? I'm guessing locally, but it would be good to be explicit.

File: docs/source/integrations/databricks_development_workflow.md [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+
+Before you import and run your Python code, you'll need to install your project's dependencies. Your project has a `requirements.txt` file for this purpose. Install the dependencies using the %pip magic command in a new cell in your notebook:
+
+```bash
+%pip install -r "/Workspace/Repos/<your_username>/iris-databricks/src/requirements.txt"
+```
+
+This command will install the dependencies listed in your requirements.txt file to the Databricks cluster attached to your notebook.
+
+#### Upload project data to DBFS
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:125]: Why is this needed? I got an error at this stage and just skipped and can still run my pipeline fine.

File: docs/source/integrations/databricks_development_workflow.md [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+
+#### Pull the new branch and check it out on Databricks Repos
+
+Navigate to your Databricks Repo in the Databricks Workspace UI. To pull the changes from your remote repository, look for the `Pull` button in the upper-right corner of the Repo page. Click it to sync the latest changes from the remote repository into your Databricks Repo.
+
+Next, check the available branches. Click on the "Branches" button in the upper-left corner of the Repo page. You should see the newly pulled branch `feat/modified-classifier` in the list. Check out the `feat/modified-classifier` branch by clicking on it in the drop-down list. Your active branch is now feat/modified-classifier.
+
+#### Re-run your project after your changes
+
+Return to your Databricks notebook. Re-run the cells you used to run your project previously. The project will now run again, giving output similar to the following:
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:278]: I think you should be explicit here about which cells need to be re-run.

**Review by merelcht — COMMENTED** [review]
Some additional comments after trying it out.

File: docs/source/integrations/databricks_development_workflow.md (hunk 2 above)
Comment (merelcht) [reply docs/source/integrations/databricks_development_workflow.md:20]: Btw, I actually tried this out on the Azure Databricks and it worked fine, so perhaps it's not necessary to specifically call out AWS things here?

Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md]: You're right about this, this was a leftover from our old documentation. I removed this in a commit that I hadn't pushed before your review.

File: docs/source/integrations/databricks_development_workflow.md (hunk 1 above)
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:4]: Fixed.

File: docs/source/integrations/databricks_development_workflow.md (hunk 7 above)
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md]: DBFS is not ephemeral, the only steps that have to be repeated are in the notebook. I will add these.

File: docs/source/integrations/databricks_development_workflow.md (hunk 10 above)
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md]: You were likely using the same cluster as me, where I have already uploaded this data to DBFS as part of my own testing.

This would be easier to explain offline.

File: docs/source/integrations/databricks_development_workflow.md (hunk 3 above)
Comment (merelcht) [reply docs/source/integrations/databricks_development_workflow.md:15]: I think it would be good to explain exactly that to the user, so they know why they need `conda` and could potentially decide to use another virtual environment if they know it has the right requirements.

File: docs/source/integrations/databricks_development_workflow.md [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+conda activate iris-databricks
+```
+
+With your Conda environment activated, install Kedro and dbx:
+
+```bash
+pip install kedro dbx --upgrade
+```
+
+dbx is an extension of the Databricks CLI, a command-line program for interacting with Databricks without using its UI. You will use dbx to sync your project's code with Databricks.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:39]:
```suggestion
dbx is an extension of the Databricks CLI, a command-line program for interacting with Databricks without using its UI. In this tutorial you will use dbx to sync your project's code with Databricks.
```

I'd also add a comment on why we recommend `dbx` to sync, exactly what you explained to me earlier: it makes iterating and experimenting faster and you don't need to commit every change you want to try out.

File: docs/source/integrations/databricks_development_workflow.md [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+
+The next step is to use dbx to sync your project to your repo. Open a new terminal instance, activate your conda environment, navigate to your project directory and start `dbx sync`:
+
+```bash
+conda activate iris-databricks
+cd <project_root>
+dbx sync repo --dest-repo iris-databricks --source . --force-include conf/local
+```
+
+This command will sync your local directory (`--source .`) with your repo (`--dest-repo iris-databricks`) on Databricks, forcing it to include the project's `conf/local` directory (`--force-include conf/local`). When started for the first time in your project directory, `dbx sync` will write output similar to the following to your terminal:
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:73]: `conf/local` could include credentials and that's why we never commit anything under that directory. I don't think we should be advising users to add that directory here, to me it feels like a potential security risk. What is the reason you explicitly add it?

File: docs/source/integrations/databricks_development_workflow.md [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+
+`dbx sync` will automatically sync any further changes made in your local project directory with your Databricks repo while it runs.
+
+```{note}
+Syncing with dbx is one-way only, meaning changes you make using the Databricks Repos code editor will not be reflected in your local environment. Only make changes to your project in your local environment while syncing, not in the editor that Databricks Repos provides.
+```
+
+### Upload project data to DBFS
+
+When run on Databricks, Kedro cannot access data stored in your project's directory. Therefore, you will need to upload your project's data to an accessible location. In this guide, we will store the data on the Databricks File System (DBFS). The PySpark Iris starter contains an environment that is set up to access data stored in DBFS (`conf/databricks`). To learn more about environments in Kedro configuration, see the [documentation](../configuration/configuration_basics.md#configuration-environments).
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:91]:
```suggestion
When run on Databricks, Kedro cannot access data stored in your project's directory. Therefore, you will need to upload your project's data to an accessible location. In this guide, we will store the data on the Databricks File System (DBFS). The PySpark Iris starter contains an environment that is set up to access data stored in DBFS (`conf/databricks`). To learn more about environments in Kedro configuration, see the [configuration documentation](../configuration/configuration_basics.md#configuration-environments).
```

To make the link more accessibility friendly.

File: docs/source/integrations/databricks_development_workflow.md [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+                    INFO     Pipeline execution completed successfully.                                      runner.py:89
+```
+
+## Modify your project and test the changes
+
+Now that your project has run successfully once, you can make changes using the convenience and power of your local development environment. In this section, you will change your project to use a modified classifier to create predictions on the iris data.
+
+### Modify the training / test split ratio
+
+The PySpark Iris starter uses a default 80-20 ratio of training data to test data when training the classifier. In this section, you will change this ratio to 70-30 by editing your project in your local environment, then run the modified project on Databricks to observe the different result.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:182]:
```suggestion
The PySpark Iris starter uses a default 80-20 ratio of training data to test data when training the classifier. In this section, you will change this ratio to 70-30 by editing your project in your local environment, then sync it with the Databricks repo using `dbx`, and then run the modified project on Databricks to observe the different result.
```

File: docs/source/integrations/databricks_development_workflow.md [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+
+```{note}
+If your cluster terminates, you must re-run your entire notebook, as libraries installed using `%pip install ...` are ephemeral. If not, repeating this step is only necessary if your project's requirements change.
+```
+
+## Summary
+
+This tutorial introduced a hybrid Kedro project development workflow on Databricks, using your local development environment, dbx, and Databricks Repos to sync code. The main steps in this workflow are:
+
+1. Creating a virtual environment and installing and configuring dbx.
```
Comment (merelcht) [inline docs/source/integrations/databricks_development_workflow.md:214]: Perhaps also add "creating a Kedro project"?

**Review by merelcht — COMMENTED** [review]
I think using `dbx` instead of `git` to sync code is a huge improvement! I left some more comments after following the guide with this new flow.

**Comment by noklam** [comment]
image

It's not a change introduce in this PR, but I only notice it now. We now have `Integrations` and `Deployment`. I didn't expect the databricks section is belong to this section. What's the plan here do we have a separate Deployment guide coming or we plan to move it to `Integrations` gradually?

File: docs/source/integrations/databricks_development_workflow.md [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+
+## Modify your project and test the changes
+
+Now that your project has run successfully once, you can make changes using the convenience and power of your local development environment. In this section, you will change your project to use a modified classifier to create predictions on the iris data.
+
+### Modify the training / test split ratio
+
+The PySpark Iris starter uses a default 80-20 ratio of training data to test data when training the classifier. In this section, you will change this ratio to 70-30 by editing your project in your local environment, then run the modified project on Databricks to observe the different result.
+
+Open the file `<project_root>/conf/base/parameters.yml` in your local environment. Edit the line `train_fraction: 0.8` to `train_fraction: 0.7` and save your changes. Look in the terminal where `dbx sync` is running, you should see it automatically sync your changes with your Databricks repo:
```
[the remark this answers is not in the document]
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:184]: Yes, see line 65:

```The next step is to use dbx to sync your project to your repo. Open a new terminal instance, activate your conda environment, navigate to your project directory and start `dbx sync`:```

Maybe this needs to be put in bold, so that the reader won't miss it.

File: docs/source/integrations/databricks_development_workflow.md [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+
+With your Conda environment activated, install Kedro and dbx:
+
+```bash
+pip install kedro dbx --upgrade
+```
+
+dbx is an extension of the Databricks CLI, a command-line program for interacting with Databricks without using its UI. You will use dbx to sync your project's code with Databricks.
+
+After installing dbx, you must authenticate the Databricks CLI with your [Databricks instance](https://docs.databricks.com/dev-tools/cli/index.html#set-up-authentication).
```
[the remark this answers is not in the document]
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:41]: Hmm, I'm not sure I follow. If you use the link included in the docs (https://docs.databricks.com/dev-tools/cli/index.html#set-up-authentication), it takes you through using `databricks configure --token` to authenticate your CLI. The Databricks docs also provides a link to the docs on how to generate a personal access token.

I don't think we should add screenshots of this process, since Databricks could change the flow for creating a PAT and we would have no automated checks for telling that our docs are out of date. This is a big problem with our current docs on Databricks and adds a lot of maintenance burden. I think we should delegate this to the Databricks docs themselves, which are guaranteed to be up-to-date and our link checker will pick up if they get moved.

I will expand the linked text and make it clear that the user should use the Databricks documentation though, since it is evidently not clear that the reader should use them right now.

[the remark this answers is not in the document]
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:41]: Hmm, I'm not sure I follow. If you use the link included on the line you are responding to (https://docs.databricks.com/dev-tools/cli/index.html#set-up-authentication), it takes you through using `databricks configure --token` to authenticate your CLI. The Databricks docs also provides a link to the docs on how to generate a personal access token.

I don't think we should add screenshots of this process, since Databricks could change the flow for creating a PAT and we would have no automated checks for telling that our docs are out of date. This is a big problem with our current docs on Databricks and adds a lot of maintenance burden. I think we should delegate this to the Databricks docs themselves, which are guaranteed to be up-to-date and our link checker will pick up if they get moved.

I will expand the linked text and make it clear that the user should use the Databricks documentation, since it is evidently not clear that the reader should use them right now.

File: docs/source/integrations/databricks_development_workflow.md (hunk 13 above)
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:73]: If the folder is missing, `session.run()` fails.

I have a feeling that `dbx sync` respects `.gitignore`, which is why it isn't included in the first place. Maybe the way around this is to engineer .gitignore so that it includes conf/local but not its contents? I can look into this, but it would mean opening a PR on our starter to fix properly.

File: docs/source/integrations/databricks_development_workflow.md [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+...
+[08/09/22 11:23:30] INFO     Model has accuracy of 0.960 on test data.                                        nodes.py:74
+                    INFO     Saving data to 'metrics' (MetricsDataSet)...                             data_catalog.py:382
+                    INFO     Completed 3 out of 3 tasks                                           sequential_runner.py:85
+                    INFO     Pipeline execution completed successfully.                                      runner.py:89
+```
+
+## Modify your project and test the changes
+
+Now that your project has run successfully once, you can make changes using the convenience and power of your local development environment. In this section, you will change your project to use a modified classifier to create predictions on the iris data.
```
[the remark this answers is not in the document]
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:178]: Good catch, thanks.

File: docs/source/integrations/databricks_development_workflow.md (hunk 13 above)
Comment (merelcht) [reply docs/source/integrations/databricks_development_workflow.md:73]: I think that actually it should be possible to do a succesful `kedro run` without a `conf/local` directory. That is a bigger change and probably needs some discussion though. For the purpose of this tutorial, could we suggest that users manually create a `conf/local` directory on the Databricks side?

Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:73]: Yeah that is potentially the easier solution, I'll see if I can make this a clear step that readers won't skip over.

File: docs/source/integrations/databricks_development_workflow.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:184]: An earlier step in the tutorial is to open a second terminal to run `dbx sync` in (line 65):

```The next step is to use dbx to sync your project to your repo. Open a new terminal instance, activate your conda environment, navigate to your project directory and start `dbx sync`:```

Since Nok had this comment also, I think it is currently not clear enough that this is a step. I will make this clearer.

File: docs/source/integrations/databricks_development_workflow.md (hunk 16 above)
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:214]: Added this.

File: docs/source/integrations/databricks_development_workflow.md [hunk 20]
```diff
@@ -0,0 +1,220 @@
...
+![Create a new notebook on Databricks](../meta/images/databricks_notebook_creation.png)
+
+### Run your project
+
+To run your project in your notebook, open it from the Databricks workspace UI and create four new cells with the following commands. Remember to replace `<databricks_username>` with your username on Databricks such that `project_root` correctly points to your project's location.
+
+1. Before you import and run your Python code, you'll need to install your project's dependencies on the cluster attached to your notebook. Your project has a `requirements.txt` file for this purpose. Add the following code to the first new cell to install the dependencies:
+
+```ipython
+%pip install -r "/Workspace/Repos/<databricks_username>/iris-databricks/src/requirements.txt"
```
[the remark this answers is not in the document]
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:161]: It's a nice idea, but I think it may be over-engineering things a bit since it is only used in two places.

File: docs/source/integrations/databricks_development_workflow.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (noklam) [reply docs/source/integrations/databricks_development_workflow.md]: Thx, the new instructions are more clear to me!

**Review by noklam — APPROVED** [review]

**Comment by jmholzer** [comment]
> I didn't expect the databricks section is belong to this section.

Which section?

> What's the plan here do we have a separate Deployment guide coming or we plan to move it to `Integrations` gradually?

We're moving all Databricks docs to 'Integrations' in the next release.

File: docs/source/integrations/databricks_development_workflow.md [hunk 21]
```diff
@@ -0,0 +1,258 @@
...
+```
+
+You can see that your model's accuracy has changed now that you are using a different classifier to produce the result.
+
+```{note}
+If your cluster terminates, you must re-run your entire notebook, as libraries installed using `%pip install ...` are ephemeral. If not, repeating this step is only necessary if your project's requirements change.
+```
+
+## Summary
+
```
Comment (stichbury) [inline docs/source/integrations/databricks_development_workflow.md:248]: I really like this summary but I actually think it would be more useful to have it at the top of the page as a mini-table of contents. Then you could link to the various sections that cover each of the elements.

I'd suggest you put it right at the top before "Prerequisites" and:
* Change the numbered bullets to symbols (because the actual text sections aren't numbered so it'll look a bit odd otherwise)
* Change the gerunds to imperative tense "Create a virtual environment..." , "Create a new Kedro project" etc. etc.

**Review by stichbury — CHANGES_REQUESTED** [review]
One revision only suggested to make the page a bit easier to navigate.

This is fantastic work @jmholzer I'm so pleased to see it.

File: docs/source/integrations/databricks_development_workflow.md (hunk 21 above)
Comment (jmholzer) [reply docs/source/integrations/databricks_development_workflow.md:259]: Excellent, I really like this suggestion too. Cheers! Made these changes in [bd10a7d](https://github.com/kedro-org/kedro/pull/2482/commits/bd10a7d449841b8aa84ae48fdb216db3e7bb39b1).

**Review by stichbury — APPROVED** [review]
⭐ ⭐ ⭐ ⭐ ⭐
